extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateEntityEnhancer.java

```diff
@@ -29,44 +32,31 @@
  */
 public final class HibernateEntityEnhancer implements BiFunction<String, ClassVisitor, ClassVisitor> {
 
-    private static final BytecodeProvider PROVIDER = new org.hibernate.bytecode.internal.bytebuddy.BytecodeProviderImpl(
+    private static final BytecodeProviderImpl PROVIDER = new org.hibernate.bytecode.internal.bytebuddy.BytecodeProviderImpl(
             ClassFileVersion.JAVA_V11);
+    private static final CoreTypePool CORE_POOL = new CoreTypePool("jakarta.", "java.", "org.hibernate.");
```

Comment:
P.S. we could add some more "framework types" here. e.g. Panache might be worth it, perhaps Jackson annotations - what else do people frequently put on entities?